cooked-validators/src/Cooked/Skeleton.hs

```diff
@@ -688,17 +688,17 @@ makeLensesFor
 
 -- | A convenience template where wallet 1 is the default signer of an
 -- otherwise empty transaction skeleton.
-txSkelTemplate :: TxSkel
-txSkelTemplate =
-  TxSkel
-    { txSkelLabel = Set.empty,
-      txSkelOpts = def,
-      txSkelMints = Map.empty,
-      txSkelValidityRange = Pl.always,
-      txSkelSigners = wallet 1 NEList.:| [],
-      txSkelIns = Map.empty,
-      txSkelOuts = []
-    }
+instance Default TxSkel where
```

Comment:
This makes sense. I guess it depends on how we understand what a `Default` instance mean. For me, it's more a matter of having a convenient element to work with for starting a continuous change process using `Tweaks`, rather than something that needs to have a well defined meaning at a conceptual level. I agree though that having `Wallet 1` sign it by default is not 100% satisfactory and might hinder the continous process by having to change it in the future, but we do have to comply with the types that constraint the list of signers to be non-empty. All in all, I would keep this definition for conveniency, while I understand the doubts you have in terms of semantics.

Reply:
Hm. I've given it more thoughts and I think this `Default` instance actually does make sense. Indeed, our skeletons are not actual transactions per se, they are our representation to eventually generate a transaction from it. So it makes sense to think of it as our default representation to eventually be transformed into a transaction. And having a default representation that does not yet contain anything and is signed by our default wallet makes sense to me, even conceptually. It would not if it were an actual transaction though.

Reply:
Some more thoughts following today's presentation: I think that our TxSkel should not force the list of signers to be non-empty. This would induce:
- the default instance would have a lot more sense, as everything would be empty by default
- from the default instance, anything could be added, and the default signer would not have to be removed if applicable
- the transaction would fail the validation process if no signer is provided, which is ok
- it could even fail before that, when we try to balance the transaction out in cooked and/or attempt to pay fees
- we will be forced to provide a signer manually and get rid once and for all of confusing default wallets and signers.